pkg/common/plugin/aws/iid.go

```diff
@@ -1,31 +1,88 @@
 package aws
 
 import (
+	"bytes"
+	"crypto/rsa"
 	"fmt"
 	"net/url"
-	"path"
+	"text/template"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/spiffe/spire/pkg/common/idutil"
+)
+
+const (
+	// PluginName for AWS IID
+	PluginName = "aws_iid"
+	// AccessKeyIDVarName env var name for AWS access key ID
+	AccessKeyIDVarName = "AWS_ACCESS_KEY_ID"
+	// SecretAccessKeyVarName env car name for AWS secret access key
+	SecretAccessKeyVarName = "AWS_SECRET_ACCESS_KEY"
 )
 
+// DefaultAgentPathTemplate is the default text/template
+var DefaultAgentPathTemplate = template.Must(template.New("agent-svid").Parse("{{ .PluginName}}/{{ .AccountID }}/{{ .Region }}/{{ .InstanceID }}"))
+
+type agentPathTemplateData struct {
+	InstanceIdentityDocument
+	PluginName  string
+	TrustDomain string
+}
+
+// IIDAgentConfig is a common config for IID Agent plugins.
+type IIDAgentConfig struct {
```

Comment:
The name is misleading, since this isn't used in the agent. Also, unifying the config here is very problematic:
1) it is not a good "common" config (has a bunch of stuff not applicable to both parties (resolver uses very little of it).
2) it mixes HCL configurables with other (exported) values which can cause parse failures and unintended setting of values.

We should limit the configuration sharing (including validation and initialization) to just those values needed to initialize the AWS session. The plugins can embed the common configuration struct in their own configuration struct.